src/payment/payment-strategy-action-creator.ts

```diff
@@ -47,8 +47,17 @@ export default class PaymentStrategyActionCreator {
                     strategy = this._strategyRegistry.get(PaymentStrategyType.NO_PAYMENT_DATA_REQUIRED);
                 }
 
-                return strategy
-                    .execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId })
+                return strategy.execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId })
+                    .catch(error => {
+                        return getGoogleRecapthaToken(store, error)
+                            .then((token: string) => {
+                                if (payload.payment) {
+                                    payload.payment.paymentRecaptchaToken = token;
+                                }
+
+                                return strategy.execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId });
```

Comment:
Do we need to recreate the order if addition actions are required for payment submission? If not, then I think the resubmission shouldn't happen at this level because `execute` usually includes a call to create an order. I remember I made a comment about consolidating the `try/catch` logic by moving the code out of `PaymentActionCreator` so you don't end up with two catch blocks. However, if order resubmission is not required, then the reverse should happen instead - the catch block here should be moved down to the `PaymentActionCreator` (and potentially `HostedForm`) instead.